pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -416,6 +416,14 @@ type HPAControllerConfiguration struct {
 	// through the kube-aggregator when enabled, instead of using the legacy metrics client
 	// through the API server proxy.
 	HorizontalPodAutoscalerUseRESTClients *bool
+	// HorizontalPodAutoscalerCPUTaintPeriod is period after pod start for which HPA will consider CPU
+	// samples from the pod contaminated by initialization and disregard them.
+	HorizontalPodAutoscalerCPUTaintPeriod metav1.Duration
+	// HorizontalPodAutoscalerInitialReadinessDelay is period after pod start during which readiness
+	// changes are treated as readiness being set for the first time. The only effect of this is that
+	// HPA will disregard CPU samples from unready pods that had last readiness change during that
+	// period.
+	HorizontalPodAutoscalerInitialReadinessDelay metav1.Duration
```

Comment:
is this actually based on the metric window?  if so, it should use the metric window reported by the custom metrics API, not be hardcoded.

Reply:
I agree in principle. But that would be significantly more complicated (currently client doesn't return metric window so I would have to change client, add more tests, and modify existing tests).

I need to merge this in 2 days so I'd rather not do it in this PR. Can we settle for a TODO. I'd send a PR addressing the TODO when I'm back from vacation (and merge it after the freeze).